src/GitVersionExe/Commands/GitVersionRootCommand.cs

```diff
@@ -0,0 +1,19 @@
+using Microsoft.Extensions.Options;
+using System.CommandLine;
+
+namespace GitVersion
+{
+    public class GitVersionRootCommand : RootCommand
+    {
+        public GitVersionRootCommand(CalculateCommand calculateCommand, IOptions<GitVersionOptions> globalOptions) : base("Versioning for your git repository, solved!")
+        {
+            // this.AddGlobalOption()
+            //this.AddGlobalOption(new Option("--target-path") { Argument = new Argument<LoggingMethod>() });
+            //this.AddGlobalOption(new Option("--logging-method") { Argument = new Argument<LoggingMethod>() });
+            GlobalOptions = globalOptions.Value;
+            this.AddCommand(calculateCommand);
```

Comment:
So rignt now we construct the command tree by adding command like this.

Now wonder if we can somehow add these commands dynamically, I mean we can register in the DI several commands and then we can resolve them in constructor as IEnumerable<Command> that will get added to the rootCommand. This will allow us to inject commands from external dll loaded as plugin/addin?

Reply:
So the idea is you have a assembly info update tool as plugin for gitversion. You define in the tool that you have a command `assemblyInfo` with it's options/arguments. Then in the bootstrapping phase of gitversion you register that command in DI. The Root command gets the list of available commands from DI and this way you have assembly info command support. Can we add this to this PR? just as a sample?

Reply:
Should we ship something "in the box", i.e. the ability to update an assemblyinfo file, but written in such a way that it can be replaced by a 3rd party plugin?  Or should we refactor it out completely, to then have the end-user configure what they want to use?

Personally, I prefer the latter, but I do worry that people will then be confused about how to go from 5.x to 6.x.

I like things being separate, which helps with the updating of those compontents.  But how do we make it so that users moving from 5.x to 6.x, have to do limited setup, configuration, to get things working how they were before?

Reply:
Well @gep13 you are right, we need to think about the migration path from 5.x to 6.0. We might want to have something in the box. Or we can have them out of the box, and somehow enable the previous behavior

Reply:
But for now we can play with the idea

Reply:
I like the latter, being able to enable the previous behaviour, just not sure what the best way to do that is ☹️

Reply:
> So rignt now we construct the command tree by adding command like this.
> 
> Now wonder if we can add these commands dynamically, I mean we can register in the DI several commands and then we can resolve them in constructor as IEnumerable that will get added to the rootCommand. This will allow us to inject commands from external dll loaded as plugin/addin.

Yep, or we can also resolve addin commands separately and add them to the root command after its constructed. The only reason for me adding the explicit command types as constructor arguments is I thought it made it a bit clearer to see how the command structure looked and would error if you forgot to register a command that we saw as core, with optional or add in commands being added after construction.